core/src/test/java/io/questdb/cutlass/line/tcp/AlterTableAttachPartitionFromSoftLinkLineTest.java

```diff
@@ -0,0 +1,194 @@
+                    Assert.assertFalse(txFile.isPartitionReadOnly(1));
+                    Assert.assertTrue(txFile.isPartitionReadOnlyByPartitionTimestamp(readOnlyPartitionTs));
+                    Assert.assertFalse(txFile.isPartitionReadOnlyByPartitionTimestamp(activePartitionTs));
+                }
+
+                // so that we know when the table writer is returned to the pool
+                final SOCountDownLatch tableWriterReturnedToPool = new SOCountDownLatch(1);
+                qdb.getCairoEngine().setPoolListener((factoryType, thread, name, event, segment, position) -> {
+                    if (Chars.equalsNc(tableName, name)) {
+                        if (factoryType == PoolListener.SRC_WRITER && event == PoolListener.EV_RETURN) {
+                            tableWriterReturnedToPool.countDown();
+                        }
+                    }
+                });
+
+                long[] timestampNano = {readOnlyPartitionTs * 1000L, activePartitionTs * 1000L};
+
+                try (LineTcpSender sender = LineTcpSender.newSender(Net.parseIPv4("127.0.0.1"), ILP_PORT, ILP_BUFFER_SIZE)) {
+                    // send O3 hitting a read-only partition
+                    for (int tick = 0; tick < 1000; tick++) {
+                        timestampNano[tick % 2] += 100_000L;
+                        sender.metric(tableName)
+                                .tag("s", "oyster")
+                                .field("l", 44)
+                                .field("i", 2023)
+                                .at(timestampNano[tick % 2]);
+                    }
+                    sender.flush();
+
+                    // send only hitting the active partition
+                    for (int tick = 0; tick < 5; tick++) {
+                        timestampNano[1] += 100_000L;
+                        sender.metric(tableName)
+                                .tag("s", "oyster")
+                                .field("l", 44)
+                                .field("i", 2023)
+                                .at(timestampNano[1]);
+                    }
+                    sender.flush();
+                }
+                tableWriterReturnedToPool.await();
+                assertSql(
+                        compiler,
+                        context,
+                        "SELECT min(ts), max(ts), count() FROM " + tableName + " sample by 1d",
+                        Misc.getThreadLocalBuilder(),
+                        "min\tmax\tcount\n" +
+                                "2022-12-08T00:05:11.070207Z\t2022-12-09T00:01:17.517546Z\t278\n" +
+                                "2022-12-09T00:06:28.587753Z\t2022-12-10T00:02:35.035092Z\t278\n" +
+                                "2022-12-10T00:07:46.105299Z\t2022-12-11T00:03:52.552638Z\t278\n" +
+                                "2022-12-11T00:09:03.622845Z\t2022-12-12T00:00:00.050500Z\t782\n"); // 505 as expected
+            }
+        });
+    }
```

Comment:
Missing test case when active partition is read-only and ILP data is in order.

Reply:
ACTIVE partition can never be read only AFAIK

Reply:
added, found an edge case, added another test Ignored, to be fixed in a follow up PR if possible